client/src/components/Bills/BillForm.js

```diff
@@ -0,0 +1,196 @@
+import { useState } from "react";
+import ContactDropDown from "../DropDowns/ContactDropList";
+
+const emptyBill = {
+  transaction_date: new Date(),
+  subtotal: 0,
+  tax_rate: 0,
+  tax_total: 0,
+  tip_rate: 0,
+  tip_total: 0,
+  who_paid: 0,
+  paid_up: null,
+  bill_notes: "",
+  location: "",
+  full_total: 0,
+};
+
+const BillForm = (props) => {
+  //An initial student if there is one in props
+  const { initialBill } = props;
+
+  // Initial State
+  const [bill, setBill] = useState(initialBill || emptyBill);
```

Comment:
I know I put 🍪 on this in other comments, but it's possible after some reading that this might introduce errors?

A more robust solution would be:
```
   const [bill , setBill] = useState(props.initialBill || emptyBill);
   useEffect(() => {
       setBill(props.initialBill);
   }, [props])
```

Which properly respond if `props` ever changes.

Reply:
`  // Initial State
  const [bill, setBill] = useState(emptyBill);

  //for the initial state, for Puts/Edits
  useEffect(() => {
    setBill(initialBill);
  }, [])

  //if there is a change in props, it updates ot
  useEffect(() => {
    setBill(props.initialBill);
  }, [props]);`